com.unity.visualeffectgraph/Runtime/Utilities/Playables/VisualEffectControl/VisualEffectControlTrackMixerBehaviour.cs

```diff
@@ -0,0 +1,760 @@
+                            ProcessEvent(chunk.clips[currentEvent.clipIndex].exit, chunk);
+                        }
+                        else if (currentEvent.clipType == Event.ClipType.Exit)
+                        {
+                            ProcessEvent(chunk.clips[currentEvent.clipIndex].enter, chunk);
+                        }
+                        //else: Ignore, we aren't playing single event backward
+                    }
+
+                    //The last event will be always invalid in case of scrubbing backward
+                    m_LastEvent = kErrorIndex;
+                }
+            }
+            else
+            {
+                var eventAhead = GetEventsIndex(chunk, oldTime, newTime, m_LastEvent);
+                foreach (var itEvent in eventAhead)
+                    ProcessEvent(itEvent, chunk);
+            }
+        }
+
+        void ProcessEvent(int eventIndex, Chunk currentChunk)
+        {
+            if (eventIndex == kErrorIndex)
+                return;
+
+            m_LastEvent = eventIndex;
+            var currentEvent = currentChunk.events[eventIndex];
+
+#if UNITY_EDITOR
+            if (currentEvent.clipType == Event.ClipType.Enter)
+            {
+                if (m_ClipState[currentEvent.clipIndex])
+                    throw new InvalidOperationException();
+
+                m_ClipState[currentEvent.clipIndex] = true;
+            }
+            else if (currentEvent.clipType == Event.ClipType.Exit)
+            {
+                if (!m_ClipState[currentEvent.clipIndex])
+                    throw new InvalidOperationException();
+
+                m_ClipState[currentEvent.clipIndex] = false;
+            }
+#endif
+            m_Target.SendEvent(currentEvent.nameId, currentEvent.attribute);
+        }
+
+        static IEnumerable<int> GetEventsIndex(Chunk chunk, double minTime, double maxTime, int lastIndex)
+        {
+            var startIndex = lastIndex == kErrorIndex ? 0 : lastIndex + 1;
+            for (int i = startIndex; i < chunk.events.Length; ++i)
+            {
+                var currentEvent = chunk.events[i];
+                if (currentEvent.time > maxTime)
+                    break;
+
+                if (minTime <= currentEvent.time)
+                    yield return i;
+            }
+        }
+
+        static VFXEventAttribute ComputeAttribute(VisualEffect vfx, EventAttributes attributes)
+        {
+            if (attributes.content == null || attributes.content.Length == 0)
+                return null;
+
+            var vfxAttribute = vfx.CreateVFXEventAttribute();
+            bool anyRelevantAttribute = false;
+            foreach (var attribute in attributes.content)
+            {
+                if (attribute != null && attribute.ApplyToVFX(vfxAttribute))
+                {
+                    anyRelevantAttribute = true;
+                }
+            }
+
+            if (!anyRelevantAttribute)
+                return null;
```

Comment:
```csharp
if (!attributes.Any(x => x?.ApplyToVFX(vfxAttribute) == true)
{
  return null;
}
```

Reply:
`Any ` would have break at the first false element, we have to apply all in this implementation.

Reply:
That's true, sorry for bad suggestion.
But this one should work as expected :)
```csharp
if (attributes.Count(x => x?.ApplyToVFX(vfxAttribute) == true) == 0)
{
  return null;
}
```

Reply:
Fixed at 0745bd0bd500cd91d7ff2c340fe558c6084480ff